src/main/kotlin/BaseModel.kt

```diff
@@ -0,0 +1,10 @@
+abstract class BaseModel(open val boardSize: Int) {
+    abstract fun canMove(turn: BaseTurn) : Pair<Boolean, Any?>
```

Comment:
Strange return type) I guess, it was "copy-pasted" from C++'s vector.find which also returns pair. You may use `Square?` instead and interpret null result as false.